remotes/docker/pusher.go

```diff
@@ -390,3 +387,16 @@ func requestWithMountFrom(req *request, mount, from string) *request {
 
 	return &creq
 }
+
+func (p dockerPusher) unexpectedResponse(ctx context.Context, resp *http.Response) error {
+	msg := "unexpected response"
+	err := errors.Errorf("%s: %s", msg, resp.Status)
+	log.G(ctx).WithField("resp", resp).Debug(msg)
+	body, bodyReadErr := ioutil.ReadAll(resp.Body)
```

Comment:
How do we feel about the readall here for an unexpected response?  This could be tricky as we could get a large response and OOM the daemon/machine. 

I think for this to be secure, we need to have some type of upper bound on this.

Reply:
I am not sure myself, this is just what I had, I was wondering if this could conditioned on the log-level somehow, but even then it could be too expensive...

Reply:
given this is just to log, seems 64kb is way more than enough? like this maybe: https://github.com/containerd/containerd/blob/master/remotes/docker/auth/fetch.go#L55